### ISSUE TITLE
Spurious CS8619 - A nullable ushort should behave identically (apart from the size of the variable) to a nullable int.

### ISSUE THREAD

**Issue by mrudat** [dotnet/roslyn#60681]
**Version Used**: 
net5.0 and net6.0 in Visual Studio 2022 as of 2022-04-11.

**Steps to Reproduce**:

```C#
using System;

#nullable enable

namespace BugReportCS8619ushort
{
    internal class Program
    {
        static void Main(string[] args)
        {
            {
                int? foo = 1;
                int? bar = 2;

                foo += bar;

                foo = foo + bar;

                Console.Write(foo);
            }

            {
                ushort? foo = 1;
                ushort? bar = 2;

#pragma warning disable CS8619 // Nullability of reference types in value doesn't match target type.
                foo += bar;
#pragma warning restore CS8619 // Nullability of reference types in value doesn't match target type.

                foo = (ushort?)(foo + bar);

                Console.Write(foo);
            }
        }
    }
}
```

**Expected Behavior**:

A nullable ushort should behave identically (apart from the size of the variable) to a nullable int.

**Actual Behavior**:

CS8619 is raised for performing math against a nullable ushort.

**Comment by jcouv** [comment]
May relate to https://github.com/dotnet/roslyn/issues/29962 (some known issues left marked in `NullableWalker.VisitCompoundAssignmentOperator`)
